### PR TITLE
OPS: Update RN

### PR DESCRIPTION
**Pull request by marcosrdz**
## Summary by CodeRabbit

- **New Features**
	- Enhanced `.gitignore` rules for better file exclusion across directories.
	- Updated C++ language standard to `c++20` and enabled Hermes for improved performance in iOS applications.
	- Added new React Native dependencies and updated existing ones for better functionality.
	- Simplified modal handling in the SendDetails component for a clearer user experience.
	- Streamlined iOS project configuration by removing unnecessary components and updating deployment target.
	- Configured Jest for better testing support with TypeScript and React Native.

- **Bug Fixes**
	- Adjusted version constraints for `cocoapods` and `activesupport` gems to improve stability.

- **Chores**
	- Upgraded various dependencies in `package.json` and build configurations to maintain compatibility with the latest features and improvements.
	- Removed obsolete localization key to refine the user interface.

### CODE REVIEW

**Comment by marcosrdz** [comment]
@Overtorment  cant figure out why e2e fails

**Comment by Overtorment** [comment]
@marcosrdz im looking but why everything else is red?

**Comment by marcosrdz** [comment]
> @marcosrdz im looking but why everything else is red?
config issue. fixed.

**Comment by Overtorment** [comment]
still failing

**Comment by Overtorment** [comment]
```
 FAIL  tests/e2e/bluewallet2.spec.js
  ● Test suite failed to run

    error TS6053: File '@react-native/typescript-config/tsconfig.json' not found.
```

File: package.json [hunk 1]
```diff
@@ -10,19 +10,23 @@
...
+    "@react-native/typescript-config": "^0.74.87",
     "@types/bip38": "^3.1.2",
     "@types/bs58check": "^2.1.0",
     "@types/create-hash": "^1.2.2",
     "@types/crypto-js": "^4.2.2",
-    "@types/jest": "^29.4.0",
+    "@types/jest": "^29.5.2",
     "@types/react": "^18.2.16",
-    "@types/react-native": "^0.72.0",
+    "@types/react-native": "^0.73.0",
```
Comment (limpbrains) [inline package.json:24]: Types are now included in RN itself

**Review by limpbrains — CHANGES_REQUESTED** [review]

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: package.json (hunk 1 above)
Comment (marcosrdz) [reply package.json]: removed

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  preset: 'react-native',
+  transform: {
+    '^.+\\.(ts|tsx)$': 'ts-jest',
+  },
+  moduleFileExtensions: ['js', 'json', 'ts', 'tsx'],
+  transformIgnorePatterns: ['node_modules/(?!((jest-)?react-native(-.*)?|@react-native(-community)?)|@rneui|silent-payments/)'],
+  setupFiles: ['./tests/setup.js'],
+  watchPathIgnorePatterns: ['<rootDir>/node_modules'],
+  setupFilesAfterEnv: ['./tests/setupAfterEnv.js'],
+};
```
Comment (Overtorment) [inline jest.config.js:11]: some stuff from `package.json`  `jest` config  got lost,  for example, i dont see

```
"testEnvironment": "<rootDir>/tests/custom-environment.js",
```


i added it recently, its part of the jest retry mechanism for our integration tests suite

File: tests/e2e/bluewallet2.spec.js [hunk 3]
```diff
@@ -208,23 +208,23 @@ describe('BlueWallet UI Tests - import BIP84 wallet', () => {
 
     // lest add another two outputs
     await element(by.id('advancedOptionsMenuButton')).tap();
-    await element(by.id('AddRecipient')).tap();
+    await element(by.text('Add Recipient')).tap();
```
Comment (Overtorment) [inline tests/e2e/bluewallet2.spec.js:211]: explain why getter by id got replaced with getter by text?

File: tsconfig.json [hunk 4]
```diff
@@ -6,14 +6,15 @@
...
     "resolveJsonModule": true,
     "module": "esnext",
+    "paths": {
+    }
   },
-  "extends": "@tsconfig/react-native/tsconfig.json",
-  "exclude": ["node_modules", "babel.config.js", "metro.config.js", "jest.config.js", "components/react-native-draggable-flatlist", "scripts/maccatalystpatches"],
-}
+  "extends": "@react-native/typescript-config/tsconfig.json",
+  "exclude": ["node_modules", "babel.config.js", "jest.config.js", "components/react-native-draggable-flatlist", "scripts/maccatalystpatches"],
```
Comment (Overtorment) [inline tsconfig.json:19]: btw do we still need to maintain mac catalyst patches ?

**Review by Overtorment — CHANGES_REQUESTED** [review]
also ios build failed

File: tests/e2e/bluewallet2.spec.js (hunk 3 above)
Comment (marcosrdz) [reply tests/e2e/bluewallet2.spec.js:211]: Cause its now accessing native UI

File: jest.config.js (hunk 2 above)
Comment (marcosrdz) [reply jest.config.js]: addressed

**Review by Overtorment — APPROVED** [review]
lgtm

**Review by limpbrains — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**